air/src/execution_step/boxed_value/stream.rs

```diff
@@ -327,6 +331,15 @@ impl fmt::Display for Stream {
     }
 }
 
+impl fmt::Display for Generation {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Generation::Nth(generation) => write!(f, "{}", generation),
+            Generation::Last => write!(f, "last"),
```

Comment:
The error below will be formatted as "stream doesn't have generation with number last, ..." that look strange.  Perhaps, Capitalizing the word would be better.